restapi/api_client.go

```diff
@@ -97,7 +97,7 @@ func NewAPIClient(opt *apiClientOpt) (*api_client, error) {
 		create_returns_object: opt.create_returns_object,
 		xssi_prefix:           opt.xssi_prefix,
 		debug:                 opt.debug,
-		redirects:             5,
+		retries:               5,
```

Comment:
A big word of caution... the API server's implementation is ultimately going to determine which HTTP methods are idempotent. I expect `GET` is just about universally idempotent, but `PUT` may or may not be depending on the implementation.

Therefore... 
- We cannot enable retry by default (since that would be for all methods)
- The decision to retry or not ought to be configurable (exposed as a parameter)
- We should consider whether it's worthwhile to hold an array of retryable methods the user can configure, too